### PR TITLE
Dispose store

### PR DESCRIPTION
**Pull request by Liaojinghui**
It is possible to crash the leveldb if the connection to the database is not closed properly.

Reproduce this issue:

run `restart` command here https://github.com/neo-project/neo-node/pull/809

### CODE REVIEW

**Comment by Liaojinghui** [comment]
Crash exception if db connection not closed properly.
```C#
System.Runtime.InteropServices.SEHException (0x80004005): External component has thrown an exception.
   at Neo.IO.Data.LevelDB.Native.leveldb_open(IntPtr options, String name, IntPtr& error)
   at Neo.IO.Data.LevelDB.DB.Open(String name, Options options)
```

**Comment by Liaojinghui** [comment]
This problem happened to me yesterday, at first, it can be recovered by restarting the program, but after dozens of times, leveldb crashed permanently, even a computer reboot can not fix it.

**Comment by erikzhang** [comment]
```
./neo-cli --repair
```

**Comment by Liaojinghui** [comment]
> ```
> ./neo-cli --repair
> ```

Tried, did not work, even tried to delete the DB folder and replace libleveldb. Only made it work again by deleting the whole project and reclone from Github.

File: src/LevelDBStore/Plugins/Storage/LevelDBStore.cs [hunk 1]
```diff
@@ -9,11 +10,25 @@ public class LevelDBStore : Plugin, IStorageProvider
...
         public override string Description => "Uses LevelDB to store the blockchain data";
 
+        private Stack<WeakReference> dbs = new();
+
         public IStore GetStore(string path)
         {
             if (Environment.CommandLine.Split(' ').Any(p => p == "/repair" || p == "--repair"))
                 DB.Repair(path, Options.Default);
-            return new Store(path);
+            IStore store = new Store(path);
```
Comment (Qiao-Jin) [inline src/LevelDBStore/Plugins/Storage/LevelDBStore.cs:19]: Dispose other references in dbs upon GetStore?

Comment (Liaojinghui) [reply src/LevelDBStore/Plugins/Storage/LevelDBStore.cs:19]: I have no idea whether those db connections are closed or not, therefore I have to keep a weak reference of them in a stack.
In this way, I can make sure that all connections are closed in Dispose with the slightest modification to existing code.

Comment (erikzhang) [reply src/LevelDBStore/Plugins/Storage/LevelDBStore.cs:19]: This method is designed for the user to be responsible for releasing the db connection.

**Pull request closed without merging**